### ISSUE TITLE
Can't get classes to compile

### ISSUE THREAD

**Issue by alexlafroscia** [esnext/es6-class#12]
No matter what I try, I can't seem to get anything to work.

Here's the command that I'm trying:

```
es6-class src/controller.js 2> error.txt
```

And here's the contents of `error.txt`

```
assert.js:92
  throw new assert.AssertionError({
        ^
AssertionError: Recursively calling visitor.visit(path) resets visitor state. Try this.visit(path) or this.traverse(path) instead.
    at Visitor.PVp.visit (/usr/local/lib/node_modules/es6-class/node_modules/recast/node_modules/ast-types/lib/path-visitor.js:107:12)
    at Object.visit (/usr/local/lib/node_modules/es6-class/node_modules/recast/node_modules/ast-types/lib/path-visitor.js:95:13)
    at transform (/usr/local/lib/node_modules/es6-class/lib/index.js:586:17)
    at NodePath.generateClassDefinitionExpression (/usr/local/lib/node_modules/es6-class/lib/index.js:423:10)
    at Context.types.PathVisitor.fromMethodsObject.visitClassDeclaration (/usr/local/lib/node_modules/es6-class/lib/index.js:80:43)
    at Context.invokeVisitorMethod (/usr/local/lib/node_modules/es6-class/node_modules/recast/node_modules/ast-types/lib/path-visitor.js:256:43)
    at Visitor.PVp.visitWithoutReset (/usr/local/lib/node_modules/es6-class/node_modules/recast/node_modules/ast-types/lib/path-visitor.js:142:21)
    at NodePath.each (/usr/local/lib/node_modules/es6-class/node_modules/recast/node_modules/ast-types/lib/path.js:96:22)
    at visitChildren (/usr/local/lib/node_modules/es6-class/node_modules/recast/node_modules/ast-types/lib/path-visitor.js:161:14)
    at Visitor.PVp.visitWithoutReset (/usr/local/lib/node_modules/es6-class/node_modules/recast/node_modules/ast-types/lib/path-visitor.js:150:9)
```

Regardless of how I try to invoke `es6-class` (as an executable or through `esnext`) I get that same error.

**Comment by asual** [comment]
This seems to be caused by recent changes in ast-types:
https://github.com/benjamn/ast-types/commit/e866485d72538ff9beee30fd0f1b542a1f242c8e

**Comment by srounce** [comment]
Had a quick shot at fixing this, but `recast` doesn't expose the methods needed to recursively visit sub-ASTs.

**Comment by joar** [comment]
+1

**Comment by caridy** [comment]
same issue here!

**Comment by eventualbuddha** [comment]
Should be fixed in v0.9.3.

**Comment by joar** [comment]
@eventualbuddha WFM. Thank you!
